openfisca_core/holders.py

```diff
@@ -112,164 +110,87 @@ def clone(self):
-            else:
-                assert returned_period.unit == u'year', \
-                    "Requested a monthly or yearly period. Got {} returned by variable {}.".format(
-                        returned_period, self.column.name)
-                intersection_months = min(requested_start_months + requested_period.size,
-                    returned_start_months + returned_period.size * 12) - requested_start_months
-                intersection_array = dated_holder.array * intersection_months / (returned_period.size * 12)
-            if array is None:
-                array = intersection_array.copy()
-            else:
-                array += intersection_array
-
-            remaining_period_months -= intersection_months
-            if remaining_period_months <= 0:
-                return self.put_in_cache(array, period, parameters.get('extra_params'))
-            if remaining_period_months % 12 == 0:
-                requested_period = requested_start.offset(intersection_months, u'month').period(u'year')
-            else:
-                requested_period = requested_start.offset(intersection_months, u'month').period(u'month')
+        return DatedHolder(self, period, array, parameters.get('extra_params'))
 
     def compute_divide(self, period = None, **parameters):
-        dated_holder = self.get_from_cache(period, parameters.get('extra_params'))
-        if dated_holder.array is not None:
-            return dated_holder
-
-        array = None
-        unit = period[0]
-        year, month, day = period.start
-        if unit == u'month':
-            # We expect the compute call to return a yearly period.
-            parameters['accept_other_period'] = True
-            dated_holder = self.compute(period = period, **parameters)
-            assert dated_holder.period.start <= period.start and period.stop <= dated_holder.period.stop, \
-                "Period {} returned by variable {} doesn't include requested period {}.".format(
-                    dated_holder.period, self.column.name, period)
-            if dated_holder.period.unit == u'month':
-                array = dated_holder.array * period.size / dated_holder.period.size
-            else:
-                assert dated_holder.period.unit == u'year', \
-                    "Requested a monthly or yearly period. Got {} returned by variable {}.".format(
-                        dated_holder.period, self.column.name)
-                array = dated_holder.array * period.size / (12 * dated_holder.period.size)
-            return self.put_in_cache(array, period, parameters.get('extra_params'))
-        else:
-            assert unit == u'year', unit
-            return self.compute(period = period)
+        # Check that the requested period matches period_behavior
+        if self.column.period_behavior != YEAR:
+            raise ValueError('Holder.compute_divide must be used on yearly variables. This is not the case for {}'.format(
```

Comment:
Same, we want to spare our users the worry of wondering what a `Holder` could be.

`Unable to divide value of rsa over time : only variables defined yearly can be divided over time`

?